src/transformers/models/t5/modeling_t5.py

```diff
@@ -318,44 +344,56 @@ def forward(
+            q_idx = torch.arange(q_len, device=device, dtype=torch.long)
+            if cache_position is not None:
+                if query_length is not None:
+                    real_seq_len_t = torch.as_tensor(query_length, device=device, dtype=torch.long)
+                else:
+                    real_seq_len_t = cache_position[-1].to(device=device, dtype=torch.long) + 1
+                start_q = real_seq_len_t.sub(q_len)  # tensor op only
+                context_pos = start_q.unsqueeze(0) + q_idx  # [q_len]
+                relative_context_positions = context_pos.view(q_len, 1)
             else:
-                position_bias = self.compute_bias(
-                    real_seq_length, key_length, device=scores.device, cache_position=cache_position
-                )
-                position_bias = position_bias[:, :, -seq_length:, :]
-
-            if mask is not None:
-                causal_mask = mask[:, :, :, : key_states.shape[-2]]
-                position_bias = position_bias + causal_mask
+                relative_context_positions = q_idx.view(q_len, 1)
 
-        position_bias_masked = position_bias
-        scores += position_bias_masked
-
-        # (batch_size, n_heads, seq_length, key_length)
-        attn_weights = nn.functional.softmax(scores.float(), dim=-1).type_as(scores)
-        attn_weights = nn.functional.dropout(attn_weights, p=self.dropout, training=self.training)
-
-        attn_output = torch.matmul(attn_weights, value_states)
+        attention_interface: Callable = eager_attention_forward
+        if self.config._attn_implementation != "eager":
+            raise ValueError(
+                "T5Attention only supports attn_implementation='eager' due to relative position bias."
+                "Got: "
+                f"{self.config._attn_implementation} instead. "
+            )
+        attn_output, attn_weights, computed_position_bias = attention_interface(
+            self,
+            query_states,
+            key_states,
+            value_states,
+            attention_mask=mask,
+            position_bias=position_bias,
+            dropout=0.0 if not self.training else self.dropout,
+            # T5 relative bias params
+            has_relative_attention_bias=self.has_relative_attention_bias,
+            is_decoder=self.is_decoder,
+            relative_attention_bias=getattr(self, "relative_attention_bias", None),
+            relative_attention_num_buckets=self.relative_attention_num_buckets,
+            relative_attention_max_distance=self.relative_attention_max_distance,
+            seq_length=seq_length,
+            relative_context_positions=relative_context_positions,
+            output_attentions=output_attentions,
+            **kwargs,
+        )
 
-        attn_output = attn_output.transpose(1, 2).contiguous()
-        attn_output = attn_output.view(batch_size, -1, self.inner_dim)
+        attn_output = attn_output.view(batch_size, -1, self.inner_dim).contiguous()
```

Comment:
```suggestion
        attn_output = attn_output.view(batch_size, seq_length, -1).contiguous()
```
Can you follow Bart more closely? Same for the view operations after the projection